index.d.ts

```diff
@@ -0,0 +1,769 @@
+        /**
+         * Read stream
+         * @type {DuplexIterableStream}
+         */
+        inboundStream: DuplexIterableStream;
+        /**
+         * Do we have a connection to read from?
+         *
+         * @type {boolean}
+         */
+        get isReadable(): boolean;
+        /**
+         * Do we have a connection to write on?
+         *
+         * @type {boolean}
+         */
+        get isWritable(): boolean;
+        /**
+         * Send a message to this peer.
+         * Throws if there is no `stream` to write to available.
+         *
+         * @param {Uint8Array} data
+         * @returns {void}
+         */
+        write(data: Uint8Array): void;
+        /**
+         * Attach a raw inbound stream and setup a read stream
+         *
+         * @param {DuplexIterableStream} stream
+         * @returns {void}
+         */
+        attachInboundStream(stream: any): void;
+        /**
+         * Attach a raw outbound stream and setup a write stream
+         *
+         * @param {Stream} stream
+         * @returns {Promise<void>}
+         */
+        attachOutboundStream(stream: any): Promise<void>;
+        /**
+         * Closes the open connection to peer
+         * @returns {void}
+         */
+        close(): void;
+    }
+    namespace PeerStreams {
+        export { DuplexIterableStream };
+    }
+    const PeerId: typeof import("peer-id");
+    /**
+     * TODO: replace with import of that type from somewhere
+     */
+    type DuplexIterableStream = any;
+    const AbortController: typeof import("abort-controller");
+    const pushable: typeof import("it-pushable");
+}
+declare module "pubsub/utils" {
+    export function randomSeqno(): Uint8Array;
+    export function msgId(from: string, seqno: Uint8Array): string;
+    export function anyMatch(a: any[] | Set<any>, b: any[] | Set<any>): boolean;
+    export function ensureArray(maybeArray: any): any[];
+    export function normalizeInRpcMessage(message: any, peerId: string): any;
+    export function normalizeOutRpcMessage(message: any): any;
+}
+declare module "pubsub/message/sign" {
+    /**
+     * Returns the PublicKey associated with the given message.
+     * If no, valid PublicKey can be retrieved an error will be returned.
+     *
+     * @param {InMessage} message
+     * @returns {Promise<PublicKey>}
+     */
+    export function messagePublicKey(message: any): Promise<any>;
+    /**
+     * Signs the provided message with the given `peerId`
+     *
+     * @param {PeerId} peerId
+     * @param {Message} message
+     * @returns {Promise<Message>}
+     */
+    export function signMessage(peerId: import("peer-id"), message: any): Promise<any>;
+    export const SignPrefix: any;
+    /**
+     * Verifies the signature of the given message
+     * @param {InMessage} message
+     * @returns {Promise<Boolean>}
+     */
+    export function verifySignature(message: any): Promise<boolean>;
+}
+declare module "pubsub/index" {
```

Comment:
the error I got from gossipsub is that `Could not find a declaration file for module 'libp2p-interfaces/src/pubsub'`. Does this mean gossipsub have to import module "pubsub/index" instead?

Reply:
I think i need to change rootDir 🤨